js/ui/paygUnlockDialog.js

```diff
@@ -81,17 +81,29 @@ var PaygUnlockCodeEntry = new Lang.Class({
         if (event.type() != Clutter.EventType.KEY_PRESS)
             return Clutter.EVENT_PROPAGATE;
 
-        let character = event.get_key_unicode();
-
-        // We only support printable characters.
-        if (!GLib.unichar_isprint(character))
-            return Clutter.EVENT_PROPAGATE;
+        let keysym = event.get_key_symbol();
+        let isDeleteKey = keysym == Clutter.KEY_Delete ||
+            keysym == Clutter.KEY_KP_Delete ||
+            keysym == Clutter.KEY_BackSpace;
+        let isEnterKey = keysym == Clutter.KEY_Return ||
+            keysym == Clutter.KEY_KP_Enter ||
+            keysym == Clutter.KEY_ISO_Enter;
+        let isExitKey = keysym == Clutter.KEY_Escape ||
+            keysym == Clutter.KEY_Tab;
+        let isArrowKey = keysym == Clutter.Left ||
+            keysym == Clutter.Right;
```

Comment:
Up and Down normally go to the beginning and end of an entry. See also: Home, End, PageUp, PageDown.

Reply:
I was trying to keep it simple and FWIW, up & down and PgUp & PgDown do not always behave that way (e.g. path bar at the top of Nautilus) so I'd rather keep those ones out.

I'll add HOME and END, though